enclave-runtime/src/lib.rs

```diff
@@ -905,54 +904,34 @@ where
 		debug!("Got following trusted calls from pool: {:?}", trusted_calls);
 	}
 
-	let mut calls = Vec::<OpaqueCall>::new();
-	let mut call_hashes = Vec::<H256>::new();
-
-	// load state before executing any calls
-	let (mut sidechain_db, state_lock) = state_handler
-		.load_for_mutation(&shard)
-		.map(|(l, s)| (SidechainDB::<SB::Block, _>::new(s), l))?;
-
-	let prev_state_hash = sidechain_db.state_hash();
-	trace!("state apriori hash: {:?}", prev_state_hash);
-
-	// update state needed for pallets
-	sidechain_db.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
-	sidechain_db.set_timestamp(&now_as_u64());
+	let execution_result = stf_executor.execute_timed_calls_batch::<PB, _>(
+		&trusted_calls,
+		latest_onchain_header,
+		&shard,
+		max_exec_duration,
+		|s| {
+			let mut sidechain_db = SidechainDB::<SB::Block, _>::new(s);
+			sidechain_db.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
+			sidechain_db.set_timestamp(&now_as_u64());
+			sidechain_db.ext
+		},
+	)?;
 
-	// retrieve trusted operations from pool
-	let trusted_calls = rpc_author.get_pending_tops_separated(shard)?.0;
+	let mut extrinsic_callbacks = execution_result.get_extrinsic_callbacks();
+	let call_hashes = execution_result
+		.get_all_execution_hashes()
+		.iter()
+		.map(|eh| eh.operation_hash)
+		.collect();
 
-	debug!("Got following trusted calls from pool: {:?}", trusted_calls);
-	// call execution
-	for trusted_call_signed in trusted_calls.into_iter() {
-		match handle_trusted_worker_call::<PB, _>(
-			&mut calls,
-			&mut sidechain_db.ext,
-			&trusted_call_signed,
-			latest_onchain_header,
-			shard,
-			on_chain_ocall,
-		) {
-			Ok(hashes) => {
-				if let Some((_, op_hash)) = hashes {
-					call_hashes.push(op_hash)
-				}
-				rpc_author
-					.remove_top(
-						vec![top_or_hash(trusted_call_signed, true)],
-						shard,
-						hashes.is_some(),
-					)
-					.unwrap();
-			},
-			Err(e) =>
-				error!("Error performing worker call (will not push top hash): Error: {:?}", e),
-		};
-		// Check time
-		if ends_at < duration_now() {
-			break
-		}
+	for executed_operation in execution_result.executed_operations.iter() {
+		rpc_author
+			.remove_top(
+				vec![executed_operation.trusted_operation_or_hash.clone()],
+				shard,
+				executed_operation.is_success(),
+			)
+			.unwrap();
```

Comment:
I know this was here before, but I think we should not panic in case we can not remove a top. Should we?

Reply:
We probably shouldn't. I'll change that to properly handle the result 👍